File: src/main/java/org/homework20241215/ramen/RamenBuilder.java
package org.homework20241215.ramen;


import org.homework20241215.shop.Item;

import java.util.HashMap;
import java.util.Map;

public class RamenBuilder {

    private String name;
    private Integer ramenPrice;
    private Map<Topping, Integer> toppings = new HashMap<>();

    public RamenBuilder setName(String name) {
        this.name = name;
        return this;
    }

    public RamenBuilder setRamenPrice(Integer ramenPrice) {
        this.ramenPrice = ramenPrice;
        return this;
    }

    public RamenBuilder addToppings(Topping topping, int toppingPrice) {
        this.toppings.put(topping, toppingPrice);
        return this;
    }

    public Ramen build() {
        return new Ramen(name, ramenPrice, toppings);
    }

}

File: src/main/java/org/homework20241222/ramen/RamenBuilder.kt
package org.homework20241222.ramen


class RamenBuilder {

    private var name: String? = null
    private var ramenPrice: Int? = null
    private val toppings: MutableMap<Topping, Int> = HashMap()

    fun setName(name: String?): RamenBuilder {
        this.name = name
        return this
    }

    fun setRamenPrice(price: Int): RamenBuilder {
        this.ramenPrice = price
        return this
    }

    fun addTopping(topping:Topping, price: Int): RamenBuilder {
        // mapに値を追加
        this.toppings += topping to price
        return this
    }

    fun build(): Ramen {
        return Ramen(name.toString(), ramenPrice.hashCode(), toppings)
    }
}

